en/middleware/micrortps.md

```diff
@@ -0,0 +1,344 @@
+# RTPS/ROS2 interface: the PX4-FastRTPS bridge
+
+This bridge adds communication capabilities between a *PX4 Autopilot* and a *Fast RTPS* application, through serial ports or
+UDP sockets, using [CDR serialization](https://en.wikipedia.org/wiki/Common_Data_Representation). The goal is to provide a DDS (Data Distribution Service) interface to PX4. This interface will also allow sharing information with the forthcoming release of ROS2 (Robot Operating System).
+
+RTPS is the underlying protocol of DDS, an standard from the OMG (Object Management Group) providing a real-time pub/sub middleware widely used in aerospace, defense and IoT applications, while in Robotics has been adopted as the middleware for ROS2.
+
+Fast RTPS implements the latest version of the RTPS protocol and a minimum DDS API, resulting in a very lightweight implementation of the standard and full access to the RTPS fine settings.
+
+![micro RTPS general scheme](../../assets/middleware/micrortps/micrortps_general_scheme.png)
+
+## Code generation
+
+Support for the functionality is mainly implemented within three new (automatically generated) code blocks:
+
+- *CDR serialization functions* are generated for specified uORB topics (those that are to be sent/received). For example, the following functions are generated for the *sensor_combined.msg*:
+
+  ```sh
+  void serialize_sensor_combined(const struct sensor_combined_s *input, char *output, uint32_t *length, struct microCDR *microCDRWriter);
+  void deserialize_sensor_combined(struct sensor_combined_s *output, char *input, struct microCDR *microCDRReader);
+  ```
+  
+  > **Note** *CDR serialization* provides a common format for exchanging serial data between different platforms.  
+
+- A *client application* that sends and receives the CDR serialized data from topics through a selected UART or UDP port.
+
+  ![micro RTPS client](../../assets/middleware/micrortps/micrortps_client.png)
```

Comment:
Image mixes too many concepts - data flow, code blocks. 

What I'd expect to see here is a diagram that either shows the inputs to the client application and the outputs (ie the the data flow within the application - serialisation, then out to uart. )
From a code/build perspective you might show the inputs/outputs of the build process. 

But what we have here is a mix of both. I don't know what you're trying to show.